gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/ios/IosColorsGenerator.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.gradle.generator.ios
+
+import com.lectra.koson.ObjectType
+import com.lectra.koson.arr
+import com.lectra.koson.obj
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.KModifier
+import dev.icerock.gradle.generator.ColorNode
+import dev.icerock.gradle.generator.ColorsGenerator
+import dev.icerock.gradle.generator.ObjectBodyExtendable
+import dev.icerock.gradle.utils.ArgbColor
+import dev.icerock.gradle.utils.parseArgbColor
+import org.gradle.api.file.FileTree
+import java.io.File
+
+@Suppress("DELEGATED_MEMBER_HIDES_SUPERTYPE_OVERRIDE")
+class IosColorsGenerator(
+    colorsFileTree: FileTree
+) : ColorsGenerator(colorsFileTree), ObjectBodyExtendable by IosGeneratorHelper() {
+    override fun getImports(): List<ClassName> {
+        return listOf(
+            ClassName("dev.icerock.moko.graphics", "Color")
+        )
+    }
+
+    override fun getClassModifiers(): Array<KModifier> = arrayOf(KModifier.ACTUAL)
+    override fun getPropertyModifiers(): Array<KModifier> = arrayOf(KModifier.ACTUAL)
+
+    override fun generateResources(resourcesGenerationDir: File, colors: List<ColorNode>) {
+        val assetsDirectory = File(resourcesGenerationDir, "Assets.xcassets")
+        assetsDirectory.mkdir()
+
+        colors.forEach { colorNode ->
+            val assetDir = File(assetsDirectory, "${colorNode.name}.colorset")
+            assetDir.mkdir()
+            val contentsFile = File(assetDir, "Contents.json")
+            contentsFile.createNewFile()
+
+            val colorContentObj = if (colorNode.isThemed()) {
+                val lightColor = parseArgbColor(colorNode.lightColor!!.toLong(16))
+                val darkColor = parseArgbColor(colorNode.darkColor!!.toLong(16))
+
+                val anyColor = buildColorIdiomJsonObj(lightColor) // set any color as light
+                val lightColorObj = buildAppearancesIdiomJsonBlock("light", lightColor)
+                val darkColorObj = buildAppearancesIdiomJsonBlock("dark", darkColor)
+
+                arr[anyColor, lightColorObj, darkColorObj]
+            } else {
+                val singleColor = parseArgbColor(colorNode.singleColor!!.toLong(16))
+
+                arr[buildColorIdiomJsonObj(singleColor)]
+            }
+
+            val resultObj = obj {
+                "colors" to colorContentObj
+                "info" to obj {
+                    "author" to "xcode"
+                    "version" to 1
+                }
+            }
```

Comment:
just use kotlinx-serialization as we use in all our projects. it support this:
```
json {
    "colors" to  colorContentObj
    "info" to json {
         "author" to "xcode"
         "version" to 1
    }
}
```

Reply:
Fixed.